lib/livebook/web_socket.ex

```diff
@@ -0,0 +1,94 @@
+defmodule Livebook.WebSocket do
+  @moduledoc false
+
+  alias Livebook.WebSocket.{
+    Client,
+    Request,
+    SessionRequest
+  }
+
+  defmodule Connection do
+    defstruct [:conn, :websocket, :ref]
+
+    @type t :: %__MODULE__{
+            conn: Client.conn(),
+            websocket: Client.websocket(),
+            ref: Client.ref()
+          }
+  end
+
+  @app_version Mix.Project.config()[:version]
+
+  @typep header :: {String.t(), String.t()}
+  @typep headers :: list(header())
+
+  @doc """
+  Connects with the WebSocket server for given URL and headers.
+  """
+  @spec connect(String.t(), headers()) :: Client.recv_fun()
```

Comment:
The return types here are different from the client because we have `%Connection{}` instead of `conn()`, right?

Reply:
You're right, good call